third_party/nvfuser/test/test_resize.cpp

```diff
@@ -0,0 +1,1594 @@
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+  auto cg_outputs = fe.runFusion(aten_inputs);
+
+  auto ref = t0.index({at::indexing::Slice(1, shape[0] - 1)});
+
+  TORCH_CHECK(ref.equal(cg_outputs[0]));
+}
+
+// Split a tensor to half and add them up
+TEST_F(NVFuserTest, FusionResizeSlice2_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({11, 30});
+
+  TORCH_CHECK(shape[1] % 2 == 0);
+
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 = slice(
+      tv0,
+      {Slice(),
+       {IrBuilder::create<Int>(0), IrBuilder::create<Int>(shape[1] / 2)}});
+  auto tv2 = slice(tv0, {Slice(), {IrBuilder::create<Int>(shape[1] / 2)}});
+  auto tv3 = add(tv1, tv2);
+  fusion.addOutput(tv3);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::manual_seed(0);
+
+  auto t0 = at::randn(shape, options);
+  std::vector<c10::IValue> aten_inputs({t0});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+  auto cg_outputs = fe.runFusion(aten_inputs);
+
+  auto t1 = t0.index(
+      {at::indexing::Slice(0, at::indexing::None),
+       at::indexing::Slice(0, shape[1] / 2)});
+  auto t2 = t0.index(
+      {at::indexing::Slice(0, at::indexing::None),
+       at::indexing::Slice(shape[1] / 2)});
+  auto ref = t1 + t2;
+
+  testValidate(&fusion, cg_outputs, aten_inputs, {ref}, __LINE__, __FILE__);
+}
+
+// "Trivial" slice is converted to Set
+TEST_F(NVFuserTest, FusionResizeSlice3_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(1);
+  fusion.addInput(tv0);
+
+  // These should result in unary set op
+  auto tv1 = slice(tv0, {{nullptr, tv0->axis(0)->extent()}});
+  auto tv2 = slice(tv0, {Slice()});
+  auto tv3 = add(tv1, tv2);
+  fusion.addOutput(tv3);
+
+  TORCH_CHECK(
+      tv1->definition()->isA<UnaryOp>() &&
+      tv1->definition()->as<UnaryOp>()->getUnaryOpType() == UnaryOpType::Set);
+  TORCH_CHECK(
+      tv2->definition()->isA<UnaryOp>() &&
+      tv2->definition()->as<UnaryOp>()->getUnaryOpType() == UnaryOpType::Set);
+}
+
+// Partition an input, reduce each and concatenate them
+TEST_F(NVFuserTest, FusionResizeSlice4_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({5, 100});
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  // Consider a fusion of:
+  // auto tv1 = add(tv0, IrBuilder::create<Double>(1));
+  // auto tv2 = sum(tv1, {1});
+
+  // Reproduce the above fusion with split tensors
+
+  // Split the input to [0:2, :] and [2:, :]
+  auto tv1 = slice(
+      tv0, {{IrBuilder::create<Int>(0), IrBuilder::create<Int>(2)}, Slice()});
+  auto tv2 = slice(tv0, {{IrBuilder::create<Int>(2)}, Slice()});
+
+  auto tv3 = add(tv1, IrBuilder::create<Double>(1));
+  auto tv4 = add(tv2, IrBuilder::create<Double>(1));
+
+  auto tv5 = sum(tv3, {1});
+  auto tv6 = sum(tv4, {1});
+  auto tv7 = cat({tv5, tv6}, 0);
+  fusion.addOutput(tv7);
```

Comment:
Are you thinking register+smem persistent kernels here??

Reply:
Yes. That's still a big gap I really wanted to fill.